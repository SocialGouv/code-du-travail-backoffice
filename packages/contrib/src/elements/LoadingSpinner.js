// https://loading.io/css/

import React from "react";
import styled, { keyframes } from "styled-components";

const Container = styled.div`
  display: inline-block;
  position: relative;
  width: ${props => props.size};
  height: ${props => props.size};
`;

const ripple = keyframes`
  0% {
    height: 0;
    left: 28px;
    opacity: 1;
    top: 28px;
    width: 0;
  }
  100% {
    height: 58px;
    left: -1px;
    opacity: 0;
    top: -1px;
    width: 58px;
  }
`;
const Spinner = styled.div`
  animation: ${ripple} 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  border-radius: 50%;
  border: 4px solid #666666;
  opacity: 1;
  position: absolute;
`;
const SpinnerNext = styled(Spinner)`
  animation-delay: -0.5s;
`;

export default ({ size }) => (
  <Container size={size !== undefined ? size : "4rem"}>
    <Spinner />
    <SpinnerNext />
  </Container>
);
