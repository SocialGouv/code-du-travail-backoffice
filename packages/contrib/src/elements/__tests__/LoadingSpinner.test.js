import React from "react";
import { render } from "react-testing-library";
import LoadingSpinner from "../LoadingSpinner";

describe("[Contrib] elements/<LoadingSpinner />", () => {
  it("should match snapshot", () => {
    const { container } = render(<LoadingSpinner size="8rem" />);

    expect(container).toMatchSnapshot();
  });
});
