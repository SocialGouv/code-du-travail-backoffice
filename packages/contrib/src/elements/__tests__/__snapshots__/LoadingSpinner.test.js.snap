// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<LoadingSpinner /> should match snapshot 1`] = `
<div>
  <div
    class="LoadingSpinner__Container-eu7nmw-0 bxJFNi"
  >
    <div
      class="LoadingSpinner__Spinner-eu7nmw-1 sHZX"
    />
    <div
      class="LoadingSpinner__Spinner-eu7nmw-1 LoadingSpinner__SpinnerNext-eu7nmw-2 daPQQA"
    />
  </div>
</div>
`;
